resilience4j-retry/src/main/java/io/github/resilience4j/retry/internal/InMemoryRetryRegistry.java

```diff
@@ -95,23 +124,38 @@ public Retry retry(String name) {
 	 */
 	@Override
 	public Retry retry(String name, RetryConfig config) {
-		return computeIfAbsent(name, () -> Retry.of(name, Objects.requireNonNull(config, CONFIG_MUST_NOT_BE_NULL)));
+		return retry(name, config, HashMap.empty());
+	}
+
+	@Override
+	public Retry retry(String name, RetryConfig config, io.vavr.collection.Map<String, String> tags) {
+		return computeIfAbsent(name, () -> Retry.of(name, Objects.requireNonNull(config, CONFIG_MUST_NOT_BE_NULL), registryTags.merge(tags)));
```

Comment:
Could you please add a `Objects.requireNonNull(tags, TAGS_MUST_NOT_BE_NULL)` here.
And maybe we should change it to `tags.merge(registryTags)`. 
The [JavaDoc ](https://static.javadoc.io/io.vavr/vavr/0.9.0/io/vavr/collection/Map.html#merge-io.vavr.collection.Map-) says: `Creates a new map which by merging the entries of this map and that map. If collisions occur, the value of this map is taken.`.
I think it's better if the instance-specific tags overwrite the registryTags, if a collision occors..

Reply:
Good catch on the `merge`!